sgx-tools/dcap_linux.go

```diff
@@ -0,0 +1,487 @@
+	if err != nil {
+		return nil, err
+	}
+
+	resp := pb.AesmServiceResponse{}
+	resp.SelectAttKeyIDRes = &pb.AesmServiceResponse_SelectAttKeyIDResponse{}
+	if err := proto.Unmarshal(rdata, &resp); err != nil {
+		return nil, err
+	}
+
+	if errCode := resp.SelectAttKeyIDRes.GetErrorCode(); errCode != 0 {
+		return nil, fmt.Errorf("failed to call selectAttKeyIDRequest (error code = %d)", errCode)
+	}
+
+	return resp.SelectAttKeyIDRes.GetSelectedAttKeyId(), nil
+}
+
+func getSupportedAttKeyIDNumRequest() (uint32, error) {
+	conn, err := dialAesmd()
+	if err != nil {
+		return 0, err
+	}
+	defer conn.Close()
+
+	req := pb.AesmServiceRequest{}
+	req.GetSupportedAttKeyIDNumReq = &pb.AesmServiceRequest_GetSupportedAttKeyIDNumRequest{
+		Timeout: aesmdTimeOut,
+	}
+
+	rdata, err := transmitAesmd(conn, &req)
+	if err != nil {
+		return 0, err
+	}
+
+	resp := pb.AesmServiceResponse{}
+	resp.GetSupportedAttKeyIDNumRes = &pb.AesmServiceResponse_GetSupportedAttKeyIDNumResponse{}
+	if err := proto.Unmarshal(rdata, &resp); err != nil {
+		return 0, err
+	}
+
+	if errCode := resp.GetSupportedAttKeyIDNumRes.GetErrorCode(); errCode != 0 {
+		return 0, fmt.Errorf("failed to call getSupportedAttKeyIDNumRequest (error code = %d)", errCode)
+	}
+
+	return resp.GetSupportedAttKeyIDNumRes.GetAttKeyIdNum(), nil
+}
+
+func getSupportedAttKeyIDsRequest(num uint32) ([]byte, error) {
+	conn, err := dialAesmd()
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+
+	req := pb.AesmServiceRequest{}
+	req.GetSupportedAttKeyIDsReq = &pb.AesmServiceRequest_GetSupportedAttKeyIDsRequest{
+		BufSize: num * attestationKeyIdentityLength,
+		Timeout: aesmdTimeOut,
+	}
+
+	rdata, err := transmitAesmd(conn, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := pb.AesmServiceResponse{}
+	resp.GetSupportedAttKeyIDsRes = &pb.AesmServiceResponse_GetSupportedAttKeyIDsResponse{}
+	if err := proto.Unmarshal(rdata, &resp); err != nil {
+		return nil, err
+	}
+
+	if errCode := resp.GetSupportedAttKeyIDsRes.GetErrorCode(); errCode != 0 {
+		return nil, fmt.Errorf("failed to call getSupportedAttKeyIDsRequest (error code = %d)", errCode)
+	}
+
+	return resp.GetSupportedAttKeyIDsRes.GetAttKeyIds(), nil
+}
+
+func getAttestationKeyIdentity(quoteType string) ([]byte, error) {
+	akId, err := selectAttKeyIDRequest()
+	if err != nil {
+		return nil, err
+	}
+	if akId == nil {
+		return nil, fmt.Errorf("failed to call selectAttKeyIDRequest")
+	}
+
+	akIdentity := &attestationKeyIdentity{}
+	if err = restruct.Unpack(akId, binary.LittleEndian, &akIdentity); err != nil {
+		return nil, err
+	}
+
+	if strings.EqualFold(quoteType, quoteTypeEcdsa) && (akIdentity.AlgorithmId == sgxQuoteLibraryAlgorithmEcdsaP256 || akIdentity.AlgorithmId == sgxQuoteLibraryAlgorithmEcdsaP384) {
+		return akId, nil
+	} else if (strings.EqualFold(quoteType, quoteTypeEpidUnlinkable) || strings.EqualFold(quoteType, quoteTypeEpidLinkable)) && akIdentity.AlgorithmId == sgxQuoteLibraryAlgorithmEpid {
+		return akId, nil
+	}
```

Comment:
Almost all code pattern like:

```go
if ... {
   ...
   return
} else if {
  // xxxx
}
```

should be converted to:

```go
if ... {
    return
}

// xxxx
```

Reply:
> Almost all code pattern like:
> 
> ```go
> if ... {
>    ...
>    return
> } else if {
>   // xxxx
> }
> ```
> 
> should be converted to:
> 
> ```go
> if ... {
>     return
> }
> 
> // xxxx
> ```

it is not a simple `if... else` but a `if ... else if ... else`.